pkg/user/tx_client_test.go

```diff
@@ -447,3 +452,241 @@ func setupEstimatorService(t *testing.T) *mockEstimatorServer {
+				mockSvc1 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					return nil, errMock1
+				}}
+				mockSvc2 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					return nil, errMock2
+				}}
+				mockSvc3 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					return nil, errMock3
+				}}
+				conn1, stop1 := grpctest.StartMockServer(t, mockSvc1)
+				conn2, stop2 := grpctest.StartMockServer(t, mockSvc2)
+				conn3, stop3 := grpctest.StartMockServer(t, mockSvc3)
+				return []*grpctest.MockTxService{mockSvc1, mockSvc2, mockSvc3}, []*grpc.ClientConn{conn1, conn2, conn3}, []func(){stop1, stop2, stop3}
+			},
+			expectedErr: errMock1, // Expect the first error returned (or any, depends on race)
+		},
+		{ // Context Deadline
+			setupMocks: func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func()) {
+				mockHandler := func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					select {
+					case <-time.After(200 * time.Millisecond):
+						return nil, errors.New("mock should have been cancelled")
+					case <-ctx.Done():
+						return nil, ctx.Err()
+					}
+				}
+				mockSvc1 := &grpctest.MockTxService{BroadcastHandler: mockHandler}
+				mockSvc2 := &grpctest.MockTxService{BroadcastHandler: mockHandler}
+				mockSvc3 := &grpctest.MockTxService{BroadcastHandler: mockHandler}
+				conn1, stop1 := grpctest.StartMockServer(t, mockSvc1)
+				conn2, stop2 := grpctest.StartMockServer(t, mockSvc2)
+				conn3, stop3 := grpctest.StartMockServer(t, mockSvc3)
+				return []*grpctest.MockTxService{mockSvc1, mockSvc2, mockSvc3}, []*grpc.ClientConn{conn1, conn2, conn3}, []func(){stop1, stop2, stop3}
+			},
+			expectedErr: context.DeadlineExceeded,
+		},
+		{ // Less Than Three Conns (Success)
+			setupMocks: func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func()) {
+				mockSvc1 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					return nil, errMock1
+				}}
+				mockSvc2 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					return &sdktx.BroadcastTxResponse{TxResponse: &sdk.TxResponse{Code: abci.CodeTypeOK, TxHash: "HASH_LT3"}}, nil
+				}}
+				conn1, stop1 := grpctest.StartMockServer(t, mockSvc1)
+				conn2, stop2 := grpctest.StartMockServer(t, mockSvc2)
+				return []*grpctest.MockTxService{mockSvc1, mockSvc2}, []*grpc.ClientConn{conn1, conn2}, []func(){stop1, stop2}
+			},
+			expectedErr: nil,
+		},
+		{ // More Than Three Conns (Success - only first 3 used)
```

Comment:
we don't need this full blown test case, we trim conns in ctor.

Reply:
ah yeah